kickstarter/android-oss
# MBL-1919: API | Implement BuildPaymentPlan Query On Checkout Screen

# 📲 What

Hooked up `BuildPaymentPlanQuery` to the checkout flow for crowdfund checkout. 

# 🤔 Why

PLOT

# 🛠 How

In crowdfund checkout, we call `buildPaymentPlanQuery` to see if the project has opted into PLOT, and pass the pledge amount to check that it meets the plot minimum. 
Then we display the ui according to the response
If the user selects plot, we pass that value back to the CrowdfundCheckoutViewModel and change the `incremental` value to true or false, based on if they user opted in or out of plot
We pass that incremental value to the `createBackingMutation`

The plot ui should only be visible on checkout for **new** pledges. 
It should **NOT** show:
- on late pledges
- on manage pledge checkout for regular pledges
- on anything other than brand **NEW** pledges

# 👀 See

Dark mode - Plot allowed on project, ineligible and eligible, unselected and selected
![plot ineligible](https://github.com/user-attachments/assets/b3f2a18c-e313-4f16-b33f-b36ae8d95628)![darkmode unexpanded](https://github.com/user-attachments/assets/6b963882-0b7b-44f6-a9c2-fd8de591af06)![darkmode expanded](https://github.com/user-attachments/assets/0659a80f-9d31-4d9d-bba0-ed1d3380483a)![plot selected](https://github.com/user-attachments/assets/82c71ab6-8a73-4465-a17e-cb690bae8e23)

Light mode - Plot allowed on project, ineligible and eligible, unselected and selected
![plot ineligible light](https://github.com/user-attachments/assets/379ee952-e2f3-4291-a668-63705e0a9d89)![plot eligible unexpanded light](https://github.com/user-attachments/assets/81f207fc-d2b9-44ac-852a-74781d53a5b8)![plot expanded light](https://github.com/user-attachments/assets/82541ec4-9052-44c1-b921-66080067a77c)![plot selected light](https://github.com/user-attachments/assets/952d4053-c805-49aa-bba4-d158627990bd)

# 📋 QA

There are a few test case scenarios. Please test these with dark and light mode, and logged in or out prior to backing:

Project = PLOT allowed, Pledge = ineligible (does not meet minimum):
- Navigate to test project "Dubstep Raspberry Pi Album"
- Go through pledge flow, set bonus amount to anything under $125
- On checkout screen, plot and pay in full should both be displayed, but PLOT option is disabled and NOT selectable
- Should show badge on plot option "Available for pledges over $125"
- All other copy should be normal pledge copy
- Hit the pledge button. Navigate to manage pledge, tap the button in the top right corner, should show all the normal manage pledge options (change reward, change payment method, etc etc)

Project = PLOT allowed, Pledge = eligible (meets the minimum)
- Navigate to test project "Dubstep Raspberry Pi Album"
- Go through pledge flow, set bonus amount to anything over $125 (previous minimum was $150, but was supposed to be changed to $125, not sure if the backend changed the minimum yet, but to be sure I usually set the bonus amount to $200)
- On checkout screen, plot and pay in full should both be displayed, but PLOT option is enabled and selectable
- Tapping on PLOT option expands the option, should see charge schedule, amounts, etc etc
- Text describing plot should be visible
- Copy displayed at bottom of screen should explain they will be charged for the first installment if the project is successful
- copy displayed at bottom should show line item badge "Please Over Time" and text "Charged in as 4 payments"
- Hit pledge, navigate to manage pledge, tap 3 dots in top right corner. Options should be limited to "Cancel Pledge" and "Contact Creator"

Project = PLOT not allowed
- Navigate to any project that is NOT "Dubstep Raspberry Pi Album"
- Go through pledge flow
- On checkout page, should not see any collection plan options. Just the regular degular checkout flow
- Pledging should

Manage Pledge = should not be able to opt into PLOT
- Back  "Dubstep Raspberry Pi Album" with a regular pay in full pledge, NOT PLOT
- Go to manage pledge, select "change payment method" or  "change reward"
- Go through checkout flow. when landing on the checkout page, should NOT have the option to opt into plot. should not display plot ui at all. 

# Story 📖

[MBL-1919: API | Implement BuildPaymentPlan Query On Checkout Screen](https://kickstarter.atlassian.net/browse/MBL-1919)
